taipy/core/data/pickle.py

```diff
@@ -79,32 +83,40 @@ def __init__(
             edit_in_progress,
             **properties,
         )
-        self.__pickle_file_path = self.__build_path()
-        if not self._last_edit_date and os.path.exists(self.__pickle_file_path):
+        self._pickle_default_path = self.__build_path()
+        if not self._last_edit_date and os.path.exists(self._pickle_default_path):
             self.unlock_edit()
-        if default_value is not None and not os.path.exists(self.__pickle_file_path):
+        if default_value is not None and not os.path.exists(self._pickle_default_path):
             self.write(default_value)
 
     @classmethod
     def storage_type(cls) -> str:
         return cls.__STORAGE_TYPE
 
-    @property
-    def path(self) -> str:
-        return self.__pickle_file_path
+    @property  # type: ignore
+    @_self_reload(_MANAGER_NAME)
+    def path(self) -> Any:
+        return self._pickle_default_path
+
+    @path.setter  # type: ignore
+    def path(self, value: Any):
+        self.properties[self.__PICKLE_DEFAULT_PATH_KEY] = value
```

Comment:
You don't need to change the default path value in properties, but the actual path. So if needed, you can retrieve what was the default path after you changed the path.
```suggestion
        self._pickle_path = value
```

Reply:
I tried this initially but it does not work because _pickle_path is a computed property, so we should set the properties to trigger the saving into JSON file.

Reply:
you can trigger the saving using the setter decorator : 
@_self_setter(_MANAGER_NAME)

Reply:
I have tried that method but it does not work. I think the problem is from the _pickle_path is a custom property itself.